src/EmailAddressValidator.php

```diff
@@ -0,0 +1,260 @@
+<?php
+
+namespace ipl\Validator;
+
+use ipl\I18n\Translation;
+
+/**
+ * Validates Email address
+ */
+class EmailAddressValidator extends BaseValidator
+{
+    use Translation;
+
+    /**
+     * @var bool If MX check should be enabled
+     */
+    protected $mx = false;
+
+    /**
+     * @var bool If a deep MX check should be done
+     */
+    protected $deep = false;
+
+    /**
+     * Create a new E-mail address validator with optional options
+     *
+     * @param bool $mx whether MX check should be enabled
+     * @param bool $deep whether a deep MX check should be done
+     */
+    public function __construct(bool $mx = false, bool $deep = false)
```

Comment:
Hm, that went a bit too far. Please check for yourself now what happens if you want to use this validator the standard way :wink: